### PR TITLE
VB-330 Additional visit support

### PR DESCRIPTION
**Pull request by philip-milne**
## What does this pull request do?
Replaces the visit free text Reasonable Adjustments column with a list of available Support.

## What is the intent behind these changes?
To all the UI to record multiple support requirements (Pre-defined & free text) against a visit.

### CODE REVIEW

**Review by andrewrlee — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/jpa/VisitSupport.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import javax.persistence.Column
+import javax.persistence.Embeddable
+import javax.persistence.Entity
+import javax.persistence.Id
+import javax.persistence.JoinColumn
+import javax.persistence.ManyToOne
+import javax.persistence.Table
+
+@Embeddable
+data class VisitSupportPk(
```
[the remark this answers is not in the document]
Comment (rj-adams) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/jpa/VisitSupport.kt:13]: I would never use a composite key - can't stand the things.

and there isn't really a need for strict referential integrity here - the selected additional support options can get copied into the visit entry

happy to discuss

[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/jpa/VisitSupport.kt:13]: I see what you mean, my comment was more about how they were using the JPA annotations, what they have done is a composite key but a obscure way of doing it that breaks some OOP rules when could be done using the composite annotation . I agree we should use use a simple indexing approach with primary keys and other keys especially when the data set is so small on this table!

**Comment by philip-milne** [comment]
Hi Aled,

Yeah I think now you're onboard it would be a good time for a clean up.
I've given both projects a light refactor for dead code and config but I've
left most of the code as is doing bits and pieces as I work on them.  I
have a list of things which work but are not ideal, maybe for next sprint
we could fix some of the important ones.

Phil

**Comment by AledWEvans** [comment]
awesome, as long as we don't break anything :-)

**Pull request merged**